### PR TITLE
change_flight_mode.py: Use the updated constant

### PR DESCRIPTION
**Pull request by refeed**
Closes https://github.com/bluerobotics/ardusub-gitbook/issues/245

### CODE REVIEW

File: developers/pymavlink/change_flight_mode.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
     ack_msg = ack_msg.to_dict()
 
     # Check if command in the same in `set_mode`
```
Comment (ES-Alexander) [inline developers/pymavlink/change_flight_mode.py:41]:
```suggestion
    # Check if acknowledged command is not `set_mode`
```

**Review by ES-Alexander — APPROVED** [review]
The example probably needs a larger revision to avoid endlessly blocking if the autopilot sends a NACK or never receives the message, in which case it could also be useful to include retries (on ACK timeout). Those points aren’t critical though, and can be handled when the pymavlink examples get [transferred over to ArduSub-Zola](https://github.com/bluerobotics/ardusub-gitbook/issues/230).

More generally, it’s best if commit messages include an indication of which directory a change is occurring in, or the topic it refers to (e.g. in this case we’d normally have `pymavlink: ` at the start of the message), but given the file name is provided it’s not vital this time.

If you’d like to make some changes given that feedback then feel free, but otherwise it can be merged as is :-)

**Comment by refeed** [comment]
Agreed, I also think it's necessary to put the code that is able to avoid endless blocking.. For now, I can only put a note on the example code about that to notify pymavlink users that that case is possible

File: developers/pymavlink/change_flight_mode.py [hunk 2]
```diff
@@ -38,8 +38,10 @@
     ack_msg = master.recv_match(type='COMMAND_ACK', blocking=True)
     ack_msg = ack_msg.to_dict()
 
-    # Check if command in the same in `set_mode`
-    if ack_msg['command'] != mavutil.mavlink.MAVLINK_MSG_ID_SET_MODE:
+    # Continue waiting if the acknowledged command is not `set_mode`
+    if ack_msg['command'] != mavutil.mavlink.MAV_CMD_DO_SET_MODE:
+        # Would be good to add mechanism to avoid endlessly blocking
+        # if the autopilot sends a NACK or never receives the message
```
Comment (ES-Alexander) [inline developers/pymavlink/change_flight_mode.py:44]: This comment should be up at line 38 where the blocking receive actually occurs.

**Review by ES-Alexander — CHANGES_REQUESTED** [review]

**Review by ES-Alexander — APPROVED** [review]
Looks good - thanks! :-)

**Pull request merged**